### PR TITLE
Remove IDropped

### PR DESCRIPTION
**Pull request by ElectroJr**
- Remove IDropped
- Moves Gas Tank logic to system
- Raises DroppedEvent in more situations
- Removes the use-only-in-hand/inventory restriction for the gas analyser
  - If this feature is desirable, should be added to #7028

### CODE REVIEW

File: Content.Shared/Interaction/SharedInteractionSystem.cs [hunk 1]
```diff
@@ -850,46 +851,14 @@ public void ThrownInteraction(EntityUid user, EntityUid thrown)
...
-            var dropMsg = new DroppedEvent(user, item);
+            var dropMsg = new DroppedEvent(user);
             RaiseLocalEvent(item, dropMsg);
             if (dropMsg.Handled)
-            {
                 _adminLogSystem.Add(LogType.Drop, LogImpact.Low, $"{ToPrettyString(user):user} dropped {ToPrettyString(item):entity}");
-                return;
-            }
-
-            Transform(item).LocalRotation = Angle.Zero;
```
Comment (mirrorcult) [inline Content.Shared/Interaction/SharedInteractionSystem.cs:880]: why was this removed?

Comment (ElectroJr) [reply Content.Shared/Interaction/SharedInteractionSystem.cs:880]: After RobustToolbox/pull/2576, entities in containers should already have zero local rotation.

Though now that I think about it RobustToolbox/pull/2594 might screw with that, seeing as it makes entities preserve world rotation when changing parents, so I guess I should add it back in.

**Review by mirrorcult — APPROVED** [review]

**Comment by mirrorcult** [comment]
conflicts then merge
